src/api/spec/controllers/webui/session_controller_spec.rb

```diff
@@ -122,4 +122,43 @@
       end
     end
   end
+
+  context 'In proxy mode' do
+    let!(:user) { create(:confirmed_user, login: 'proxy_user') }
+    let(:username) { 'new_user' }
+
+    before do
+      # Fake proxy mode
+      stub_const('CONFIG', CONFIG.merge('proxy_auth_mode' => :on))
+    end
+
+    it 'logs in a user when the header is set' do
+      request.headers['HTTP_X_USERNAME'] = user.login
+
+      # a rather unusual place to go, but this isn't really
+      # about the session controller but about basic proxy mode
+      get :new
+      expect(User.current).to eq(user)
+    end
+
+    it 'does not log in any user when no header is set' do
+      get :new
+      expect(User.current.login).to eq '_nobody_'
+    end
+
+    it 'creates a new user account if user does not exist in OBS' do
+      # check we're fine
```

Comment:
why? how can that not be fine?

Reply:
@Ana06 - this are machines. If you trust the machines, you're part of the problem!

![](http://www.willtheterminatorcometrue.com/wp-content/uploads/2015/12/terminator-2-t-1000-shake.gif)

Reply:
Funny comments are not so funny anymore when you come back to them 3 years later and have to figure out what they mean...

Reply:
Spaßbremse!

Reply:
well... I don't trust this comment... how could it be that we are not fine? can you please elaborate? :thinking: